src/index.ts

```diff
@@ -60,9 +60,9 @@ export class PromiseQueue {
 				}
 				let [timeoutSeconds] = process.hrtime();
```

Comment:
To keep this with `process.hrtime.bigint()` all the way:
```suggestion
				let timeoutNano = process.hrtime.bigint();
				timeoutNano += BigInt(maxAgeSeconds) * 1000000000n;
```
(you can't do 1e9n sadly, although having written it down it does look a little crazy)

Reply:
Compiler error: `BigInt literals are not available when targeting lower than ESNext.`.

I pushed instead `BigInt(maxAgeSeconds * 1e9)`, since that's several orders of magnitude smaller than `Number.MAX_SAFE_INTEGER`:

```
> 30 * 1e9
30000000000
> Number.MAX_SAFE_INTEGER
9007199254740991
```